Objects/longobject.c

```diff
@@ -4309,18 +4315,34 @@ long_rshift(PyLongObject *a, PyLongObject *b)
     }
     else {
         shiftby = PyLong_AsSsize_t((PyObject *)b);
-        if (shiftby == -1L && PyErr_Occurred())
-            return NULL;
-        if (shiftby < 0) {
-            PyErr_SetString(PyExc_ValueError,
-                            "negative shift count");
-            return NULL;
+        if (shiftby >= 0) {
+            wordshift = shiftby / PyLong_SHIFT;
+            loshift = shiftby - wordshift * PyLong_SHIFT;
```

Comment:
> I suppose this is because multiplication and substraction are faster than the "%" operation.

This seems unlikely to me. In particular, I'd expect an optimising compiler to be able to recognise and do a good job of optimising the pair `shiftby / PyLong_SHIFT` and `shiftby % PyLong_SHIFT` if they appear next to each other in code. Rewriting the `%` would likely make it harder for the compiler to recognise.